cli/command/plugin/install.go

```diff
@@ -39,52 +43,82 @@ func newInstallCommand(dockerCli *command.DockerCli) *cobra.Command {
 	flags := cmd.Flags()
 	flags.BoolVar(&options.grantPerms, "grant-all-permissions", false, "Grant all permissions necessary to run the plugin")
 	flags.BoolVar(&options.disable, "disable", false, "Do not enable the plugin on install")
+	flags.StringVar(&options.alias, "alias", "", "Local tag reference for plugin")
 
 	return cmd
 }
 
-func runInstall(dockerCli *command.DockerCli, opts pluginOptions) error {
-	named, err := reference.ParseNamed(opts.name) // FIXME: validate
+func getRepoIndex(ref distreference.Named) (*registrytypes.IndexInfo, error) {
+	named, err := reference.ParseNamed(ref.Name())
 	if err != nil {
-		return err
+		return nil, err
 	}
-	if reference.IsNameOnly(named) {
-		named = reference.WithDefaultTag(named)
+
+	repoInfo, err := registry.ParseRepositoryInfo(named)
+	if err != nil {
+		return nil, err
 	}
-	ref, ok := named.(reference.NamedTagged)
-	if !ok {
-		return fmt.Errorf("invalid name: %s", named.String())
+
+	return repoInfo.Index, nil
+}
+
+func runInstall(dockerCli *command.DockerCli, opts pluginOptions) error {
+	ref, err := distreference.ParseNamed(opts.name)
```

Comment:
Yuck :(. Is this using `distreference` to avoid the normalization? Needs a huge FIXME to explain why `distreference` and what should be done when the packages are merged.